applications/IgaApplication/custom_modelers/refinement_modeler.cpp

```diff
@@ -172,6 +172,88 @@ namespace Kratos
                             << " by inserting knots in v, however insert_nb_per_span_v is set to 0." << std::endl;
                     }
                 }
+                if (rParameters["parameters"].Has("increase_degree_u")) {
+                    SizeType degree_u = rParameters["parameters"]["increase_degree_u"].GetInt();
+
+                    if (degree_u > 0) {
+                        std::vector<double> spans_local_space;
+                        p_nurbs_surface->SpansLocalSpace(spans_local_space, 0);
+
+                        KRATOS_INFO_IF("::[RefinementModeler]::ApplyRefinement", mEchoLevel > 1)
+                            << "Refining nurbs surface #" << p_nurbs_surface->Id() << " by elevating degree in u: "
+                            << degree_u << std::endl;
+
+                        PointerVector<NodeType> PointsRefined;
+                        Vector KnotsURefined;
+                        Vector WeightsRefined;
+
+                        NurbsSurfaceRefinementUtilities::DegreeElevationU(*(p_nurbs_surface.get()), degree_u,
+                            PointsRefined, KnotsURefined, WeightsRefined);
+
+                        // Recreate nodes in model part to ensure correct assignment of dofs
+                        IndexType node_id = (r_model_part.NodesEnd() - 1)->Id() + 1;
+                        for (IndexType i = 0; i < PointsRefined.size(); ++i) {
+                            if (PointsRefined(i)->Id() == 0) {
+                                PointsRefined(i) = r_model_part.CreateNewNode(node_id, PointsRefined[i][0], PointsRefined[i][1], PointsRefined[i][2]);
+                                node_id++;
+                            }
+                        }
+
+                        KRATOS_INFO_IF("::[RefinementModeler]::ApplyRefinement", mEchoLevel > 1)
+                            << "New knot vector: " << KnotsURefined
+                            << ", new weights vector: " << WeightsRefined << std::endl;
+
+                        p_nurbs_surface->SetInternals(PointsRefined,
+                            p_nurbs_surface->PolynomialDegreeU() + degree_u, p_nurbs_surface->PolynomialDegreeV(),
+                            KnotsURefined, p_nurbs_surface->KnotsV(),
+                            WeightsRefined);
+                    } else {
+                        KRATOS_INFO_IF("::[RefinementModeler]::ApplyRefinement", mEchoLevel > 1)
+                            << "Trying to refine nurbs surface #" << p_nurbs_surface->Id()
+                            << " by elevating degree in u, however increase_degree_u is set to 0." << std::endl;
+                    }
+                }
+                if (rParameters["parameters"].Has("increase_degree_v")) {
+                    SizeType degree_v = rParameters["parameters"]["increase_degree_v"].GetInt();
```

Comment:
I think this variable name could be also confusing

Reply:
thanks for pointing this out